pkg/controller/spinnakerservice/spinnakerservice_controller.go

```diff
@@ -139,10 +140,10 @@ func (r *ReconcileSpinnakerService) Reconcile(request reconcile.Request) (reconc
 			return reconcile.Result{Requeue: true}, nil
 		}
 	}
-	sc := newStatusChecker(r.client, reqLogger, TypesFactory)
+	sc := newStatusChecker(r.client, reqLogger, TypesFactory, r.evtRecorder, util.NewK8sLookup(r.client))
 	if err = sc.checks(instance); err != nil {
 		r.evtRecorder.Eventf(instance, corev1.EventTypeWarning, "StatusError", "Error updating SpinnakerService status: %s", err.Error())
-		return reconcile.Result{}, err
+		return reconcile.Result{Requeue: true}, err
```

Comment:
You shouldn't need to requeue. If an error is returned, the operator framework will do a retry with back-off to avoid infinite loops.